### PR TITLE
Add support for configuration file with release-public value

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/builder/issues/72

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
RFR @giantswarm/cena @denderello

File: commands/builder.go [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const (
+	ConfigFileName   = "builder"
+	ConfigFileFormat = "json"
```
Comment (xh3b4sd) [inline commands/builder.go]: Why no yaml config here? We started to use that in the other projects too.

**Comment by xh3b4sd** [comment]
The description states this PR fixes issue #72. It does not. This PR is about public release. The issue is about multiple repos.

File: commands/release.go [hunk 2]
```diff
@@ -20,6 +22,9 @@ func init() {
 	releaseCommand.PersistentFlags().StringVar(&privateRegistry, "docker-private-registry", "registry.giantswarm.io", "URL of the private docker registry")
 	releaseCommand.PersistentFlags().StringVar(&privateRegistryNamespace, "docker-private-namespace", "giantswarm", "user/company namespace within the private registry")
 	releaseCommand.PersistentFlags().StringVar(&publicRegistry, "docker-public-registry", "giantswarm", "URL of the public docker registry")
+
+	viper.BindPFlag("release-public", releaseCommand.PersistentFlags().Lookup("release-public"))
+	releasePublicDockerImage = viper.GetBool("release-public")
```
Comment (xh3b4sd) [inline commands/release.go]: Cool when this works now. In the past it did not. Was buggy. Whatever.

**Comment by xh3b4sd** [comment]
Fix the issue reference and go ahead as you like. LGTM

File: commands/builder.go (hunk 1 above)
Comment (zeisss) [inline commands/builder.go]: why even do this at all? Viper supports just picking whatever is there, json/yml/toml etc. We don't need to pin this down to one format.

Comment (JosephSalisbury) [inline commands/builder.go]: Because I'm being dumb - fix incoming

**Pull request merged**